modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/service/internal/search/index/test/InstanceWorkflowMetricsIndexerTest.java

```diff
@@ -115,8 +128,86 @@ public void testReindex() throws Exception {
 			workflowDefinition.getWorkflowDefinitionId());
 	}
 
+	@Test
+	public void testUpdateAssigneeUser() throws Exception {
+		KaleoInstance kaleoInstance = getKaleoInstance(addBlogsEntry());
+
+		retryAssertCount(
+			booleanQuery -> booleanQuery.addMustQueryClauses(
+				queries.nested(
+					"tasks",
+					queries.term("tasks.assigneeType", Role.class.getName()))),
+			1,
+			_instanceWorkflowMetricsIndexNameBuilder.getIndexName(
+				workflowDefinition.getCompanyId()),
+			"WorkflowMetricsInstanceType", "className",
+			kaleoInstance.getClassName(), "classPK", kaleoInstance.getClassPK(),
+			"companyId", kaleoInstance.getCompanyId(), "completed", false,
+			"deleted", false, "instanceId", kaleoInstance.getKaleoInstanceId(),
+			"processId", workflowDefinition.getWorkflowDefinitionId(),
+			"version", "1.0");
+
+		User user = UserTestUtil.addUser(
+			RandomTestUtil.randomString(
+				NumericStringRandomizerBumper.INSTANCE,
+				UniqueStringRandomizerBumper.INSTANCE),
+			LocaleUtil.getDefault(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(),
+			new long[] {TestPropsValues.getGroupId()});
+
+		List<KaleoTaskInstanceToken> kaleoTaskInstanceTokens =
+			_kaleoTaskInstanceTokenLocalService.getKaleoTaskInstanceTokens(
+				kaleoInstance.getKaleoInstanceId(), false, 0, 1, null,
+				ServiceContextTestUtil.getServiceContext(
+					TestPropsValues.getGroupId(), 0L));
+
+		assignKaleoTaskInstanceToken(
+			kaleoInstance, kaleoTaskInstanceTokens.get(0), user.getUserId());
+
+		retryAssertCount(
+			booleanQuery -> booleanQuery.addMustQueryClauses(
+				queries.nested(
+					"tasks",
+					queries.term("tasks.assigneeName", user.getFullName()))),
+			1,
+			_instanceWorkflowMetricsIndexNameBuilder.getIndexName(
+				workflowDefinition.getCompanyId()),
+			"WorkflowMetricsInstanceType", "className",
+			kaleoInstance.getClassName(), "classPK", kaleoInstance.getClassPK(),
+			"companyId", kaleoInstance.getCompanyId(), "completed", false,
+			"deleted", false, "instanceId", kaleoInstance.getKaleoInstanceId(),
+			"processId", workflowDefinition.getWorkflowDefinitionId(),
+			"version", "1.0");
+
+		user.setMiddleName(RandomTestUtil.randomString());
+
+		_userLocalService.updateUser(user);
+
```

Comment:
Hey @MarinhoFeliphe please reassign the `user` variable or create a new one to receive the result of the `updateUser` call;